src/Microsoft.AspNetCore.SignalR.Core/HubConnectionList.cs

```diff
@@ -32,14 +32,29 @@ public void Remove(HubConnectionContext connection)
             _connections.TryRemove(connection.ConnectionId, out _);
         }
 
-        public IEnumerator<HubConnectionContext> GetEnumerator()
+        public Enumerator GetEnumerator()
         {
-            return _connections.Values.GetEnumerator();
+            return new Enumerator(this);
         }
 
-        IEnumerator IEnumerable.GetEnumerator()
+        public struct Enumerator : IEnumerator<HubConnectionContext>
         {
-            return GetEnumerator();
+            private IEnumerator<KeyValuePair<string, HubConnectionContext>> _enumerator;
```

Comment:
readonly? Also can the struct be readonly?

Reply:
struct based enumerators are usually mutable structs but this *might* since it's just wrapping the dictionary enumerator.

Reply:
Was it not `readonly`able?

Reply:
Works